src/pipecat/processors/aggregators/llm_response.py

```diff
@@ -109,6 +110,7 @@ async def _push_aggregation(self):
 
             # Reset
             self._aggregation = ""
+            self._aggregating = False
             self._seen_start_frame = False
             self._seen_end_frame = False
             self._seen_interim_results = False
```

Comment:
We have a bug where we append the previous user speech a second time, because the "Reset" block here comes after the await self.push_frame()

Example messages from the buggy case:

  {
    "role": "user",
    "content": " Tell me a long story about a horse."
  },
  {
    "role": "assistant",
    "content": " Certainly!"
  },
  {
    "role": "user",
    "content": " Tell me a long story about a horse. Oh, wait. I changed my mind. I want you to"
  },
  
```suggestion
            # Reset our accumulator state
            self._aggregation = ""
            self._aggregating = False
            self._seen_start_frame = False
            self._seen_end_frame = False
            self._seen_interim_results = False
            # Push the frame
            await self.push_frame(frame)
```

Reply:
BUT NOTE THAT THE SUGGESTION ABOVE IS INCOMPLETE. ALSO NEED TO REMOVE LINE 102. (I can't figure out how to make that "suggestion" in the github ux.)

Reply:
Since frames are in order, the current `process_frame` call is the one that calls this `push_frame`. A following call to `process_frame` can't happen yet because we need to finish this `push_frame` until we reach last processor downstream. So, it seems where we do the reset doesn't really matter? But I might be missing something.

Reply:
Oh! Fomr the spoken words, this looks like an interruption. If so, yes, there could be a bug. I believe we should probably do the reset when we receive a `StartInterruptionFrame`. I'll investigate.

Reply:
Ah, OK! I know why the ordered mattered. The thing is that with interruptions the task is cancelled. So the code after `push_frame` is not executed. So, moving it before would fix it. However, it's better to use `StartInterruptionFrame` as I suggested.